app/models/internal/event.rb

```diff
@@ -0,0 +1,121 @@
+module Internal
+  class Event
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :id, :string, default: nil
+    attribute :readable_id, :string
+    attribute :status_id, :integer
+    attribute :name, :string
+    attribute :summary, :string
+    attribute :description, :string
+    attribute :is_online, :boolean
+    attribute :start_at, :datetime
+    attribute :end_at, :datetime
+    attribute :provider_contact_email, :string
+    attribute :provider_organiser, :string
+    attribute :provider_target_audience, :string
+    attribute :provider_website_url, :string
+    attribute :building
+
+    validates :name, presence: true, allow_blank: false
+    validates :readable_id, presence: true, allow_blank: false
+    validates :summary, presence: true, allow_blank: false
+    validates :description, presence: true, allow_blank: false
+    validates :summary, presence: true, allow_blank: false
+    validates :is_online, inclusion: { in: [true, false] }
+    validates :start_at, presence: true, allow_blank: false
+    validates :end_at, presence: true
+    validates :provider_contact_email,
+              presence: true,
+              allow_blank: false,
+              email_format: true,
+              length: { maximum: 100 }
+    validates :provider_organiser, presence: true, allow_blank: false
+    validates :provider_target_audience, presence: true, allow_blank: false
+    validates :provider_website_url, presence: true, allow_blank: false
+    validates_each :start_at, :end_at do |record, attr, value|
+      unless value.nil?
+        record.errors.add attr, "Must be in the future" if value <= Time.zone.now
+      end
+    end
+    validate :end_after_start
+
+    def persisted?
+      id.present?
+    end
+
+    def submit_pending
+      self.status_id = GetIntoTeachingApiClient::Constants::EVENT_STATUS["Pending"]
+      submit
+    end
+
+    def approve
+      self.status_id = GetIntoTeachingApiClient::Constants::EVENT_STATUS["Open"]
+      submit
+    end
+
+  private
+
+    def submit
+      return false if invalid?
+
+      submit_to_api?
+    end
+
+    def end_after_start
+      return if end_at.blank? || start_at.blank?
+
+      if end_at.floor <= start_at.floor
+        errors.add(:end_at, "must be after the start date")
+      end
+    end
+
+    def submit_to_api?
```

Comment:
I think having this in the model is muddying the waters a bit. I would be inclined to have the event/api_event mapping logic in the model but the submission either in the controller or a separate class that handles internal event submissions, This also feels very verbose; I think the mapping to/from an internal event to an api event could be more dynamic (similar to what you have in the `transform_event` controller method already). It feels odd its partly done in the controller one way and then the model the other way; needs making consistent.